integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,468 @@
+		assertThat(updatedTask.getStatus(), is(Task.TaskStatus.ACCEPTED));
+	}
+	
+		@Test
+	public void shouldDeleteExistingTask() throws Exception {
+		MockHttpServletResponse response = delete("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		
+		response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, statusEquals(HttpStatus.GONE));
+	}
+	
+	@Test
+	public void shouldReturnNotFoundWhenDeletingNonExistentTask() throws Exception {
+		MockHttpServletResponse response = delete("/Task/" + WRONG_TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldSearchForExistingTasksAsJson() throws Exception {
+		MockHttpServletResponse response = get("/Task").accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Bundle results = readBundleResponse(response);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.getType(), equalTo(Bundle.BundleType.SEARCHSET));
+		assertThat(results.hasEntry(), is(true));
+		
+		List<Bundle.BundleEntryComponent> entries = results.getEntry();
+		
+		assertThat(entries, everyItem(hasProperty("fullUrl", startsWith("http://localhost/ws/fhir2/R3/Task/"))));
+		assertThat(entries, everyItem(hasResource(instanceOf(Task.class))));
+		assertThat(entries, everyItem(hasResource(validResource())));
+	}
+	
+	@Test
+	public void shouldSearchForExistingTaskAsXml() throws Exception {
+		MockHttpServletResponse response = get("/Task").accept(FhirMediaTypes.XML).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Bundle results = readBundleResponse(response);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.getType(), equalTo(Bundle.BundleType.SEARCHSET));
+		assertThat(results.hasEntry(), is(true));
+		
+		List<Bundle.BundleEntryComponent> entries = results.getEntry();
+		
+		assertThat(entries, everyItem(hasProperty("fullUrl", startsWith("http://localhost/ws/fhir2/R3/Task/"))));
+		assertThat(entries, everyItem(hasResource(instanceOf(Task.class))));
+		assertThat(entries, everyItem(hasResource(validResource())));
+		
+	}
+	
+	@Test
+ public void shouldReturnSortedAndFilteredSearchResultsForTaskAsJson() throws Exception {
+	   MockHttpServletResponse response = get("/Task/?status=accepted&_sort=_lastUpdated").accept(FhirMediaTypes.JSON).go();
```

Comment:
@ibacher , this is what i added to test the filtered search , hope this is what you meant

Reply:
That's exactly the kind of thing I was hoping for!

Reply:
Hey @ibacher, i thought you had suggested this change /Task/?status=accepted&_sort=_lastUpdated". which i had added before. Kindly review this because it had taken alot of time and it seems to ready to go thanks

Reply:
@ibacher thats great  thanks so we are ready to merge this !